test/Riok.Mapperly.Tests/Mapping/RestrictedKeywordTest.cs

```diff
@@ -0,0 +1,63 @@
+namespace Riok.Mapperly.Tests.Mapping;
+
+public class RestrictedKeywordTest
+{
+    [Fact]
+    public void NewInstance()
```

Comment:
If the proposed approach works, I think a simple test in `Riok.Mapperly.Tests.Mapping.MapperTest` should be sufficient.